### PR TITLE
Catch Parsing Error of Datetime

### LINKED ISSUE #3483: keyword search formatted wrong throws error
https://sentry.io/organizations/codalab/issues/2387369760/

better error cataloging/don't throw errors when they should be user-facing

### PR DESCRIPTION
**Pull request by jzwang43**
### Reasons for making this change

isoparse is throwing an unhandled error if an invalid date is provided, triggering sentry.io

### Related issues

fixes #3483 

### Screenshots

image

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

File: codalab/model/bundle_model.py [hunk 1]
```diff
@@ -529,7 +529,10 @@ def make_condition(key, field, value):
                         alias(select([cl_worksheet_item.c.bundle_uuid]).where(condition))
                     )
             elif key in ('.before', '.after'):
-                target_datetime = parser.isoparse(value)
+                try:
+                    target_datetime = parser.isoparse(value)
+                except Exception:
+                    raise UsageError("Unable to parse datatime, please check the format.")
```
Comment (epicfaace) [inline codalab/model/bundle_model.py]:
```suggestion
                    raise UsageError("Unable to parse datetime. Datetime must be specified as an ISO-8601 datetime string such as YYYY-MM-DD.")
```

File: codalab/model/bundle_model.py [hunk 2]
```diff
@@ -529,7 +529,10 @@ def make_condition(key, field, value):
                         alias(select([cl_worksheet_item.c.bundle_uuid]).where(condition))
                     )
             elif key in ('.before', '.after'):
-                target_datetime = parser.isoparse(value)
+                try:
+                    target_datetime = parser.isoparse(value)
+                except Exception:
```
Comment (epicfaace) [inline codalab/model/bundle_model.py]: more specific exception

**Review by epicfaace — CHANGES_REQUESTED** [review]

File: codalab/model/bundle_model.py (hunk 1 above)
Comment (jzwang43) [reply codalab/model/bundle_model.py]: Updated.

File: codalab/model/bundle_model.py (hunk 2 above)
Comment (jzwang43) [reply codalab/model/bundle_model.py]: Changed to `ValueError` which is thrown by `isoparse`.

**Review by epicfaace — APPROVED** [review]

**Pull request merged**